geospaas_processing/downloaders.py

```diff
@@ -99,8 +99,22 @@ def get_remote_file_size(response, auth):
         return file_size
 
     @classmethod
-    def download_url(cls, url, download_dir, file_prefix='', **kwargs):
-        """Download file from HTTP URL"""
+    def get_file_name(cls, response, url, file_prefix=None):
+        """ Get the name of downloaded file either from header of <response>, or from <url>.
+        Prepend the file name with <file_prefix> if given """
+        file_name = cls.extract_file_name(response)
+        if not file_name:
+            file_name = os.path.basename(urlparse(url).path).rstrip('.dods')
```

Comment:
Why do you need to remove dods here? It should not be in the URL that you save in the database. In the database there should be a URL for downloading netCDF file. It always ends with `.nc` not with `.dods`.

Reply:
Actually, on some OpenDAP servers, all download URLs end with `.dods` (DODS is the name of the binary download service).

However, there are still some issues:
- as mentioned in another PR, using `os.path.basename()` on a URL is not a great practice, because depending on the OS the path separator might not be `/`.
- I don't like ".dods" being hard coded here. Maybe something could be done with the `provider_settings.yml` file?
- the names of the `get_file_name()` and `extract_file_name()`  methods are too close to each other. Maybe rename `extract_file_name()` to `extract_file_name_from_response()`?